Mathlib/Algebra/Algebra/Rat.lean

```diff
@@ -25,25 +26,89 @@ theorem map_rat_algebraMap [Semiring R] [Semiring S] [Algebra ℚ R] [Algebra 
 
 end RingHom
 
-section Rat
+namespace NNRat
+variable [DivisionSemiring R] [CharZero R]
+
+section Semiring
+variable [Semiring S] [CharZero S] [NNRatCast S] [Module ℚ≥0 S]
+
+variable (R) in
+/-- `nnqsmul` is equal to any other module structure via a cast. -/
+lemma cast_smul_eq_nnqsmul [Module R S] (q : ℚ≥0) (a : S) : (q : R) • a = q • a := by
+  refine MulAction.injective₀ (α := ℚ≥0) (Nat.cast_ne_zero.2 q.den_pos.ne') ?_
+  dsimp
+  rw [← mul_smul, den_mul_eq_num, Nat.cast_smul_eq_nsmul, Nat.cast_smul_eq_nsmul, ← smul_assoc,
+    nsmul_eq_mul q.den, ← cast_natCast, ← cast_mul, den_mul_eq_num, cast_natCast,
+    Nat.cast_smul_eq_nsmul]
+
+end Semiring
+
+section DivisionSemiring
+variable [DivisionSemiring S] [CharZero S]
+
+instance instAlgebra : Algebra ℚ≥0 R where
+  smul_def' := smul_def
+  toRingHom := castHom _
+  commutes' := cast_commute
+
+instance instLinearMapClass [FunLike F R S] [RingHomClass F R S] :
```

Comment:
Arguably, `RingHomClass.toNNRatLinearMapClass` would be a more accurate name.

Reply:
Yes, true